File: subprojects/webapp/src/app/services/event-type.service.ts
import {Injectable} from 'angular2/core';

import {Http} from 'angular2/http';

import * as Rx from 'rxjs';

import {EventType} from '../model/model';



@Injectable()
export class EventTypeService {

    constructor(private http: Http) { }

    getAll(): Rx.Observable<Array<EventType>> {
        return this.http.get('/api/eventtypes')
            .map(res => res.json());
    }
}

File: src/main/java/com/github/twostone/leaderboard/services/EventTypeService.java
package com.github.twostone.leaderboard.services;

import com.github.twostone.leaderboard.model.event.EventType;
import com.github.twostone.leaderboard.model.event.EventTypeManager;

import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RestController;

import javax.inject.Inject;

@RestController
@RequestMapping("api/eventtypes")
public class EventTypeService {
  
  private EventTypeManager eventTypeManager;
  
  @Inject
  EventTypeService(EventTypeManager eventTypeManager) {
    super();
    this.eventTypeManager = eventTypeManager;
  }

  @RequestMapping(
      path = "", 
      method = RequestMethod.GET)
  Iterable<EventType> getEventTypes() {
    return this.eventTypeManager.getAll();
  }
}
